lib/internal/fs/cp/cp-sync.js

```diff
@@ -64,7 +63,7 @@ function cpSyncFn(src, dest, opts) {
 function checkPathsSync(src, dest, opts) {
   if (opts.filter) {
     const shouldCopy = opts.filter(src, dest);
-    if (isPromise(shouldCopy)) {
+    if (typeof shouldCopy !== 'boolean') {
```

Comment:
I have the feeling this was done on purpose, to support truthy values, and there's a Promise check to prevent mistakes.

Reply:
https://github.com/nodejs/node/pull/39372#pullrequestreview-724939078

Reply:
Thanks for your feedback. and I have a question, for 
```javascript
opt.filter = (value) => {
 //
}
```
its return type contradicts the [documentation](https://nodejs.org/docs/latest/api/fs.html#fscpsyncsrc-dest-options), which states `Returns: <boolean>`, but it wont throw an error. Should here use a more precise description, such as `the type that is coercible to boolean type`?

Reply:
/cc @nodejs/fs @bcoe @aduh95

Reply:
I'm fine with either (changing the code or the docs); if we change the code, let's land this as semver-major.

Reply:
I think I might prefer changing the docs to changing the code here.